recommender/api/endpoints/recommendations.py

```diff
@@ -30,11 +30,24 @@ def post(self):
 
         json_dict = service_ctx(request.get_json())
         engine, engine_name = load_engine(json_dict)
+        panel_id = json_dict.get("panel_id")
         try:
-            services_ids = engine(json_dict)
+            services_ids, scores, explanations = engine(json_dict)
+            explanations_long, explanations_short = [
+                list(t) for t in list(zip(*[(e.long, e.short) for e in explanations]))
+            ]
             deserialize_recommendation(json_dict, services_ids, engine_name)
 
-            response = {"recommendations": services_ids}
+            response = {
```

Comment:
According to review, I've added "engine_version" field. However, I'm wondering if it is necessary because now the response contains exactly the same field "engine_version" like in the /recommendations request body. Why somebody would ask for recommendations from some specific "engine_version" and then get "engine_version" as a part of the response? It doesn't provide any new information for the requester. What do you think @Michal-Kolomanski ?

Reply:
Good that you are vigilant! But the requested engine is not always used to serve recommendations. For example, we are requesting an NCF engine, but it is not trained (as well as RL), so recommendations are served using the random engine.

Reply:
BTW. your "engine_version" field name it's better

Reply:
Good point that "engine_version" is meaningful information when recommender falls back to the default engine. So we stay with this field in the response.